tracing/providers/opentracing/src/main/java/io/helidon/tracing/providers/opentracing/OpenTracingTracer.java

```diff
@@ -106,6 +106,9 @@ public <T> T unwrap(Class<T> tracerClass) {
         if (tracerClass.isAssignableFrom(delegate.getClass())) {
             return tracerClass.cast(delegate);
         }
+        if (tracerClass.isInstance(this)) {
```

Comment:
Minor nit: depending on how "deep" you want/need the unwrapping to go you could do this whole general thing in a `while` loop